### ISSUE TITLE
What is `this` context suppose to be?

### ISSUE THREAD

**Issue by chemoish** [creynders/dijon#6]
I am trying to work with webpack (module loader) and dash.js.

I noticed that version `0.5.3` is still being used by their latest master, but i am finding that there are issues with dijon.

If you include this file in the correct context, the `window`, then it will work. However, if you include this in a different context, the module loader's context, then it doesn't work.

Is there a reason why this isn't just `window` (If that indeed is what it is expecting)?

https://github.com/creynders/dijon/blob/master/src/dijon.js#L29
https://github.com/creynders/dijon/blob/v0.5.3/src/dijon.js#L618

---

Maybe webpack just isn't compatible with this implementation.

Thanks.

FYI: Webpack passes in a context of `{}`.

**Comment by creynders** [comment]
@chemoish hiya, if I remember correctly (but it's a looong time ago and so much has changed in the meantime) I just copy pasted it from one of the big libs (but can't remember which one) since I didn't have a lot of experience working with module loaders.

That said, I don't understand the problem fully, you're using webpack, but you need dijon as a global object?? I haven't used dijon with webpack yet, I'll try it out as soon as I've got time.

**Comment by chemoish** [comment]
While, I don't really have much experience with AMD or UMD, or dijon TBH. It seems dijon follows the UMD format of...

```js
(function (root, factory) {
  // ...
}(this, function () {
  // ...
}))
```

In the case of Video.js, it uses dijon and expects `this` to be the context of the window. If it's anything other than the `window`, dijon won't work.

Module loader like webpack wraps code with something like.

```js
// i am making this up but it gets the idea across
function load() {
  // videojs code
  
  // dijon code
}

load.apply({});
```

The context would have been `window`, but the context was manually set to `{}` by webpack. Which I guess goes back to the original conclusion is that probably not compatible—webpack supports `AMD` and `CommonJS`.

**Comment by creynders** [comment]
That's actually the whole idea of using something like webpack: encapsulating code and not exposing it globally. However, Video.js probably uses the globally exposed dijon object, in which case you need to configure webpack correctly: http://webpack.github.io/docs/shimming-modules.html
Should be something like this:

```js
new webpack.ProvidePlugin({
    dijon: "dijon",
    "window.dijon": "dijon"
})
```

**Comment by chemoish** [comment]
Makes sense. Thanks for the help.

**Comment by chemoish** [comment]
I got this working, but it seems there are still issues with the encapsulation.

```js
new webpack.ProvidePlugin({
    dijon: "dijon",
    "window.dijon": "dijon"
})
```

First, `window.dijon` does nothing (or so it seems).

Second, `dijon` wraps any code which needs to access `dijon` by exporting `dijon` to a module, then that module is included in said code.

```js
// does something similar
(function (dijon) {
  // videojs
}(__webpack_require__(5))); // assume export 5 is npm dijon
```

This makes videojs rely upon the dijon included by npm (`0.6.1`)—Seems to work. Instead of relying on the one that was embedded inside the videojs code.

#fwiw
